src/getLocationForJsonPath.ts

```diff
@@ -99,6 +109,38 @@ function findNodeAtPath(node: YAMLNode, path: JsonPath, closest: boolean) {
   return node;
 }
 
+function getMappings(mappings: YAMLMapping[], mergeKeys: boolean): YAMLMapping[] {
+  if (!mergeKeys) return mappings;
+
+  return mappings.reduce<YAMLMapping[]>((mergedMappings, mapping) => {
+    if (isValidNode(mapping as YAMLNode)) {
+      if (mapping.key.value === SpecialMappingKeys.MergeKey) {
+        mergedMappings.push(...reduceMergeKeys((mapping as YAMLMapping).value as YAMLNode));
+      } else {
+        mergedMappings.push(mapping);
+      }
+    }
+
+    return mergedMappings;
+  }, []);
+}
+
+function reduceMergeKeys(node: YAMLNode): YAMLMapping[] {
+  switch (node.kind) {
+    case Kind.SEQ:
+      return (node as YAMLSequence).items.reduceRight<YAMLMapping[]>((items, item) => {
+        items.push(...reduceMergeKeys(item));
+        return items;
+      }, []);
+    case Kind.MAP:
+      return node.mappings;
+    case Kind.ANCHOR_REF:
+      return reduceMergeKeys((node as YAMLAnchorReference).value);
```

Comment:
These casts are so weird — TypeScript supports tagged union, these should not be necessary :(

Reply:
It's my oversight - it's not necessary. Thanks for spotting.